types/src/key_prefix.rs

```diff
@@ -0,0 +1,231 @@
+#[cfg(any(feature = "testing", test))]
+use crate::testing::TestRng;
+use crate::{
+    account::AccountHash,
+    bytesrepr::{self, FromBytes, ToBytes, U8_SERIALIZED_LENGTH},
+    contract_messages::TopicNameHash,
+    system::{auction::BidAddrTag, mint::BalanceHoldAddrTag},
+    EntityAddr, KeyTag, URefAddr,
+};
+use alloc::vec::Vec;
+#[cfg(any(feature = "testing", test))]
+use rand::Rng;
+
+/// Key prefixes used for querying the global state.
+#[derive(Debug, PartialEq, Eq, Clone, PartialOrd, Ord, Hash)]
+pub enum KeyPrefix {
+    /// Retrieves all delegator bid addresses for a given validator.
+    DelegatorBidAddrsByValidator(AccountHash),
+    /// Retrieves all messages for a given entity.
+    MessagesByEntity(EntityAddr),
+    /// Retrieves all messages for a given entity and topic.
+    MessagesByEntityAndTopic(EntityAddr, TopicNameHash),
+    /// Retrieves all named keys for a given entity.
+    NamedKeysByEntity(EntityAddr),
+    /// Retrieves all gas balance holds for a given purse.
+    GasBalanceHoldsByPurse(URefAddr),
+    /// Retrieves all processing balance holds for a given purse.
+    ProcessingBalanceHoldsByPurse(URefAddr),
+}
```

Comment:
i don't think this should be put in / published with `casper_types`; it just becomes yet another leaked implementation detail that causes breaking changes when we extend it / change it / go with a different implementation.

I would be more ok with it being put in casper_storage

Reply:
I've added 2 versions of the enum, one in `storage` and one in `binary_port` and removed this one